### ISSUE TITLE
make sure protractor report is generated when tests crash

### ISSUE THREAD

**Issue by jludvice** [syndesisio/syndesis-ui#559]
in come cases tests fail and crash protractor.
As a result of this crash we dont' have test report.

obvious problems like this are covered - we have html report with proper error message 
`TypeError: Cannot read property 'listComponent' of null`
```ts
    const a : IntegrationsListPage =null;
    a.listComponent();
```

**Comment by jludvice** [comment]
First candidate to fix: searching for missing dom element
```
  ✔ When Camilla selects the "Twitter Example" connection
2017-05-25 14:59:33,318 INFO [root] searching for integration action 'Mention'
[14:59:33] E/launcher - No element found using locator: By(css selector, syndesis-list-actions)
[14:59:33] E/launcher - NoSuchElementError: No element found using locator: By(css selector, syndesis-list-actions)
    at WebDriverError (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/selenium-webdriver/lib/error.js:27:5)
    at NoSuchElementError (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/selenium-webdriver/lib/error.js:168:5)
    at elementArrayFinder.getWebElements.then (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/protractor/lib/element.ts:851:17)
    at ManagedPromise.invokeCallback_ (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/selenium-webdriver/lib/promise.js:1366:14)
    at TaskQueue.execute_ (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/selenium-webdriver/lib/promise.js:2970:14)
    at TaskQueue.executeNext_ (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/selenium-webdriver/lib/promise.js:2953:27)
    at asyncRun (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/selenium-webdriver/lib/promise.js:2813:27)
    at /home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/selenium-webdriver/lib/promise.js:676:7
    at process._tickCallback (internal/process/next_tick.js:109:7)Error
    at ElementArrayFinder.applyAction_ (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/protractor/lib/element.ts:482:23)
    at ElementArrayFinder.(anonymous function) [as click] (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/protractor/lib/element.ts:96:21)
    at ElementFinder.(anonymous function) [as click] (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/protractor/lib/element.ts:873:14)
    at ListActionsComponent.selectAction (/home/jludvice/devel/redhat/ipaas/ipaas-ui/e2e/integrations/edit/edit.po.ts:54:79)
    at IntegrationSteps.selectIntegrationAction (/home/jludvice/devel/redhat/ipaas/ipaas-ui/e2e/integrations/integration.steps.ts:56:10)
    at World.bindingFunc (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/cucumber-tsflow/src/BindingDecorator.ts:145:85)
[14:59:33] E/launcher - Process exited with error code 199
```

**Comment by jludvice** [comment]
and here's code to blame
```ts
  @when(/^Camilla selects the "([^"]*)" integration.*$/)
  public selectConnection(itegrationName: string, callback: CallbackStepDefinition): void {
    const page = new IntegrationsListPage();

    page.listComponent().goToIntegration(itegrationName)
      .then(()=> callback());
  }
```

need to update it to catch possible Promise rejection
```ts
   page.listComponent().goToIntegration(itegrationName)
      .then(()=> callback())
      .catch(e => callback(e)); // call callback when sth fails and pass error
```

**Comment by jludvice** [comment]
as a result of this crash we leave chrome browser open
https://github.com/angular/protractor/issues/4178

**Comment by jludvice** [comment]
Some issues fixed by PR #560.
I don't want to do big refactor now.
We'll fix other situations as we go.
Closing issue.
